extensions/common/vault/vault-hashicorp/src/main/java/org/eclipse/edc/vault/hashicorp/HashicorpVaultTokenRenewTask.java

```diff
@@ -0,0 +1,131 @@
+/*
+ *  Copyright (c) 2023 Mercedes-Benz Tech Innovation GmbH
+ *
+ *  This program and the accompanying materials are made available under the
+ *  terms of the Apache License, Version 2.0 which is available at
+ *  https://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  SPDX-License-Identifier: Apache-2.0
+ *
+ *  Contributors:
+ *       Mercedes-Benz Tech Innovation GmbH - Implement automatic Hashicorp Vault token renewal
+ *
+ */
+
+package org.eclipse.edc.vault.hashicorp;
+
+import org.eclipse.edc.spi.monitor.Monitor;
+import org.eclipse.edc.spi.system.ExecutorInstrumentation;
+import org.jetbrains.annotations.NotNull;
+
+import java.util.concurrent.Executors;
+import java.util.concurrent.Future;
+import java.util.concurrent.ScheduledExecutorService;
+import java.util.concurrent.TimeUnit;
+import java.util.concurrent.atomic.AtomicBoolean;
+
+/**
+ * This task implements the Hashicorp Vault token renewal mechanism.
+ * To ensure that this task is really cancelled, call the stop method before program shut down.
+ */
+public class HashicorpVaultTokenRenewTask {
+
+    private static final String INITIAL_TOKEN_LOOK_UP_ERR_MSG_FORMAT = "Initial token look up failed with reason: %s";
+    private static final String INITIAL_TOKEN_RENEW_ERR_MSG_FORMAT = "Initial token renewal failed with reason: %s";
+    private static final String SCHEDULED_TOKEN_RENEWAL_ERR_MSG_FORMAT = "Scheduled token renewal failed: %s";
+
+    @NotNull
+    private final ExecutorInstrumentation executorInstrumentation;
+    @NotNull
+    private final HashicorpVaultClient client;
+    @NotNull
+    private final Monitor monitor;
+    private final long renewBuffer;
+    private ScheduledExecutorService scheduledExecutorService;
+    private Future<?> tokenRenewTask;
+    private final AtomicBoolean isRunning = new AtomicBoolean(false);
+
+    public HashicorpVaultTokenRenewTask(@NotNull ExecutorInstrumentation executorInstrumentation,
```

Comment:
at least to me it is not obvious what the intended use of `renewBuffer` is or what values I should pass. Please add javadoc to the CTor